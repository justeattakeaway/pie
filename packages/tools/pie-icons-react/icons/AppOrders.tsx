import * as React from "react";
import { SVGProps } from "react";

const AppOrders = (props: SVGProps<SVGSVGElement>) => <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 28 28" className="c-pieIcon c-pieIcon--appOrders" width="1em" height="1em" role="img" {...props}><path d="M7 2.625H10.22L10.78 4.0425H15.47L16.03 2.625H19.25C19.4821 2.625 19.7046 2.71719 19.8687 2.88128C20.0328 3.04538 20.125 3.26794 20.125 3.5V5.25H21.875V3.5C21.875 2.80381 21.5984 2.13613 21.1062 1.64384C20.6139 1.15156 19.9462 0.875 19.25 0.875H7C6.30381 0.875 5.63613 1.15156 5.14384 1.64384C4.65156 2.13613 4.375 2.80381 4.375 3.5V24.5C4.375 25.1962 4.65156 25.8639 5.14384 26.3562C5.63613 26.8484 6.30381 27.125 7 27.125H8.75V25.375H7C6.76794 25.375 6.54538 25.2828 6.38128 25.1187C6.21719 24.9546 6.125 24.7321 6.125 24.5V3.5C6.125 3.26794 6.21719 3.04538 6.38128 2.88128C6.54538 2.71719 6.76794 2.625 7 2.625Z" /><path d="M16.205 19.25H20.545L21.875 17.9987V14.875H20.125V17.5H16.625V14.875H14.875V17.9987L16.205 19.25Z" /><path d="M25.5762 11.375H21.875V8.75L20.2037 7H16.5462L14.875 8.75V11.375H11.1737L10.5612 24.85C10.5483 25.1434 10.5947 25.4363 10.6979 25.7113C10.801 25.9863 10.9586 26.2376 11.1613 26.4501C11.364 26.6626 11.6075 26.8319 11.8773 26.9479C12.1471 27.0639 12.4376 27.1241 12.7312 27.125H24.0187C24.3124 27.1241 24.6029 27.0639 24.8726 26.9479C25.1424 26.8319 25.386 26.6626 25.5887 26.4501C25.7914 26.2376 25.949 25.9863 26.0521 25.7113C26.1553 25.4363 26.2017 25.1434 26.1887 24.85L25.5762 11.375ZM16.625 8.75H20.125V11.375H16.625V8.75ZM24.325 25.2437C24.2856 25.2854 24.2381 25.3185 24.1854 25.341C24.1328 25.3636 24.076 25.3752 24.0187 25.375H12.7312C12.6739 25.3752 12.6172 25.3636 12.5645 25.341C12.5119 25.3185 12.4644 25.2854 12.425 25.2437C12.3476 25.1575 12.3068 25.0445 12.3112 24.9287L12.845 13.125H23.905L24.4387 24.9287C24.4432 25.0445 24.4024 25.1575 24.325 25.2437Z" /></svg>;

export default AppOrders;