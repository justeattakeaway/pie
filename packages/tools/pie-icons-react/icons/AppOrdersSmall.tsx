import * as React from "react";
import { SVGProps } from "react";

const AppOrdersSmall = (props: SVGProps<SVGSVGElement>) => <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 14 14" className="c-pieIcon c-pieIcon--appOrdersSmall" width="1em" height="1em" role="img" {...props}><path d="M2.40625 1.75V12.25C2.40625 12.308 2.4293 12.3637 2.47032 12.4047C2.51134 12.4457 2.56698 12.4688 2.625 12.4688H4.375V13.7812H2.625C2.2196 13.779 1.83145 13.6169 1.54478 13.3302C1.25811 13.0436 1.09605 12.6554 1.09375 12.25V1.75C1.09605 1.3446 1.25811 0.95645 1.54478 0.669781C1.83145 0.383112 2.2196 0.221047 2.625 0.21875H9.625C10.0304 0.221047 10.4186 0.383112 10.7052 0.669781C10.9919 0.95645 11.154 1.3446 11.1562 1.75V3.5H9.84375V1.75C9.84375 1.69198 9.8207 1.63634 9.77968 1.59532C9.73866 1.5543 9.68302 1.53125 9.625 1.53125H7.875L7.4725 2.485H4.7775L4.375 1.53125H2.625C2.56698 1.53125 2.51134 1.5543 2.47032 1.59532C2.4293 1.63634 2.40625 1.69198 2.40625 1.75ZM12.9062 12.25C12.918 12.4575 12.8877 12.6653 12.817 12.8608C12.7463 13.0563 12.6368 13.2355 12.495 13.3875C12.3844 13.5095 12.2493 13.6069 12.0985 13.6732C11.9478 13.7396 11.7847 13.7734 11.62 13.7725H6.74625C6.58154 13.7734 6.41849 13.7396 6.26773 13.6732C6.11697 13.6069 5.98187 13.5095 5.87125 13.3875C5.73062 13.235 5.62242 13.0556 5.55324 12.8601C5.48405 12.6645 5.4553 12.457 5.46875 12.25L5.7925 7.20125H7V5.90625L8.30375 4.6025H10.0538L11.375 5.90625V7.21875H12.5825L12.9062 12.25ZM8.3125 7.21875H10.0625V5.90625H8.3125V7.21875ZM11.5938 12.32L11.375 8.53125H7L6.78125 12.3375C6.76513 12.3888 6.76513 12.4437 6.78125 12.495H11.5238C11.559 12.4422 11.5829 12.3826 11.5938 12.32Z" /></svg>;

export default AppOrdersSmall;