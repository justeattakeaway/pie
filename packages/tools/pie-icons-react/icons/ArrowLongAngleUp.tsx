import * as React from "react";
import { SVGProps } from "react";

const ArrowLongAngleUp = (props: SVGProps<SVGSVGElement>) => <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 28 28" className="c-pieIcon c-pieIcon--arrowLongAngleUp" width="1em" height="1em" role="img" {...props}><path d="M23.2838 22.0412L7.81375 6.57123H17.7363V4.82123H6.58C6.11587 4.82123 5.67075 5.0056 5.34256 5.33379C5.01438 5.66198 4.83 6.1071 4.83 6.57123V17.7362H6.58V7.80498L22.05 23.275L23.2838 22.0412Z" /></svg>;

export default ArrowLongAngleUp;