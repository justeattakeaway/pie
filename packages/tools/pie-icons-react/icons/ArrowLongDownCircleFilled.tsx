import * as React from "react";
import { SVGProps } from "react";

const ArrowLongDownCircleFilled = (props: SVGProps<SVGSVGElement>) => <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 28 28" className="c-pieIcon c-pieIcon--arrowLongDownCircleFilled" width="1em" height="1em" role="img" {...props}><path d="M14.875 1.79376V17.1413L19.04 12.9763L20.2738 14.21L15.2338 19.25C14.9059 19.576 14.4623 19.7589 14 19.7589C13.5377 19.7589 13.0941 19.576 12.7663 19.25L7.72626 14.21L8.96001 12.9763L13.125 17.1413V1.79376C9.96001 2.02041 7.00644 3.46633 4.88619 5.82707C2.76593 8.18782 1.64444 11.2792 1.75783 14.4503C1.87123 17.6213 3.21067 20.6247 5.49414 22.828C7.7776 25.0312 10.8269 26.2625 14 26.2625C17.1731 26.2625 20.2224 25.0312 22.5059 22.828C24.7893 20.6247 26.1288 17.6213 26.2422 14.4503C26.3556 11.2792 25.2341 8.18782 23.1138 5.82707C20.9936 3.46633 18.04 2.02041 14.875 1.79376Z" /></svg>;

export default ArrowLongDownCircleFilled;