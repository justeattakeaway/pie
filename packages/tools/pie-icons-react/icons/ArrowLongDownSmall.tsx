import * as React from "react";
import { SVGProps } from "react";

const ArrowLongDownSmall = (props: SVGProps<SVGSVGElement>) => <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 16 16" className="c-pieIcon c-pieIcon--arrowLongDownSmall" width="1em" height="1em" role="img" {...props}><path d="M7.24999 1V12.65L3.48999 8.89L2.42999 10L7.11999 14.68C7.23525 14.7961 7.37235 14.8883 7.52338 14.9511C7.67441 15.014 7.83639 15.0464 7.99999 15.0464C8.16359 15.0464 8.32557 15.014 8.47661 14.9511C8.62764 14.8883 8.76473 14.7961 8.87999 14.68L13.57 10L12.51 8.89L8.74999 12.65V1H7.24999Z" /></svg>;

export default ArrowLongDownSmall;