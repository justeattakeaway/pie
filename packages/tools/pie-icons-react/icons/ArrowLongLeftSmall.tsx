import * as React from "react";
import { SVGProps } from "react";

const ArrowLongLeftSmall = (props: SVGProps<SVGSVGElement>) => <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 16 16" className="c-pieIcon c-pieIcon--arrowLongLeftSmall" width="1em" height="1em" role="img" {...props}><path d="M15 7.25005H3.35L7.11 3.49005L6.05 2.43005L1.37 7.12005C1.2539 7.23531 1.16175 7.37241 1.09887 7.52344C1.03599 7.67447 1.00362 7.83645 1.00362 8.00005C1.00362 8.16365 1.03599 8.32563 1.09887 8.47667C1.16175 8.6277 1.2539 8.76479 1.37 8.88005L6.05 13.5701L7.11 12.5101L3.35 8.75005H15V7.25005Z" /></svg>;

export default ArrowLongLeftSmall;