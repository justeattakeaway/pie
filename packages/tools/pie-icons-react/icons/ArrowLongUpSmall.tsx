import * as React from "react";
import { SVGProps } from "react";

const ArrowLongUpSmall = (props: SVGProps<SVGSVGElement>) => <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 14 14" className="c-pieIcon c-pieIcon--arrowLongUpSmall" width="1em" height="1em" role="img" {...props}><path d="M7.65625 13.125V2.93126L10.9463 6.22126L11.8738 5.29376L7.77 1.19876C7.66915 1.09717 7.54919 1.01654 7.41704 0.961522C7.28488 0.906501 7.14315 0.878174 7 0.878174C6.85685 0.878174 6.71512 0.906501 6.58296 0.961522C6.45081 1.01654 6.33085 1.09717 6.23 1.19876L2.12625 5.29376L3.05375 6.22126L6.34375 2.93126V13.125H7.65625Z" /></svg>;

export default ArrowLongUpSmall;