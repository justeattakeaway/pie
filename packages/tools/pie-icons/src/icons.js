import Icon from './icon';
import icons from '../dist/icons.json'; // eslint-disable-line import/no-unresolved

export default Object.keys(icons)
  .map(key => new Icon(key, icons[key].contents, icons[key].attrs, icons[key].pathPrefix))
  .reduce((object, icon) => {
      object[icon.name] = icon;
      return object;
  }, {});
