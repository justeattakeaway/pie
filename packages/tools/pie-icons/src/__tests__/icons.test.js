/* eslint-env jest */
import icons from '../icons';

jest.mock('../../dist/icons.json', () => ({
    icon1: {
        contents: '<line x1="23" y1="1" x2="1" y2="23" /><line x1="1" y1="1" x2="23" y2="23" />',
    },
    icon2: {
        contents: '<circle cx="12" cy="12" r="11" />',
    },
}));

test('exports correct object', () => {
    expect(icons).toMatchSnapshot();
});
