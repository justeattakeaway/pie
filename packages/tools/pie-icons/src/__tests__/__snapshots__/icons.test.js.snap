// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`exports correct object 1`] = `
{
  "icon1": Icon {
    "attrs": {
      "fill": "currentColor",
      "focusable": false,
      "role": "presentation",
      "xmlns": "http://www.w3.org/2000/svg",
    },
    "contents": "<line x1=\\"23\\" y1=\\"1\\" x2=\\"1\\" y2=\\"23\\" /><line x1=\\"1\\" y1=\\"1\\" x2=\\"23\\" y2=\\"23\\" />",
    "name": "icon1",
  },
  "icon2": Icon {
    "attrs": {
      "fill": "currentColor",
      "focusable": false,
      "role": "presentation",
      "xmlns": "http://www.w3.org/2000/svg",
    },
    "contents": "<circle cx=\\"12\\" cy=\\"12\\" r=\\"11\\" />",
    "name": "icon2",
  },
}
`;
